### PR TITLE
Use Issue Forms instead of templates

### PR DESCRIPTION
**Pull request by droidmonkey**
Updates our Issue templates to Issue Forms and uses the type field instead of applying labels. Also updated the PR template a little (note that PR's don't support forms yet).

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+  - type: dropdown
+    id: os
+    attributes:
+      label: Operating System
+      description: Select your operating system.
+      options:
+        - Windows
+        - Linux
+        - macOS
```
Comment (phoerious) [inline .github/ISSUE_TEMPLATE/bug_report.yml:57]: Maybe add Other for the BSDs etc.

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        - Other
+
+  - type: dropdown
+    id: window_system
+    attributes:
+      label: Linux Windowing System
+      description: If on Linux, please select your windowing system.
+      options:
+        - X11
+        - Wayland
```
Comment (phoerious) [inline .github/ISSUE_TEMPLATE/bug_report.yml:81]: This should already be in the debug info. Let's not make it too tedious.

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+type: Bug
+
+body:
+  - type: checkboxes
+    attributes:
+      label: Have you searched for an existing issue?
+      description: Use the issue search box to see if one already exists for the bug you encountered.
+      options:
+      - label: Yes, I tried searching
+        required: true
```
Comment (phoerious) [inline .github/ISSUE_TEMPLATE/bug_report.yml:14]: Perhaps also mention the pinned issues.

File: .github/ISSUE_TEMPLATE/feature_request.yml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    id: example
+    attributes:
+      label: Example
+      description: Provide an example of how this feature would be used.
+
+  - type: textarea
+    id: context
+    attributes:
+      label: Context
+      description: Why does this feature matter to you? What unique circumstances do you have?
```
Comment (phoerious) [inline .github/ISSUE_TEMPLATE/feature_request.yml:33]: This sounds like a field where people will get "creative". Not sure what the informational value to us will be.

File: .github/ISSUE_TEMPLATE/bug_report.yml (hunk 2 above)
Comment (droidmonkey) [reply .github/ISSUE_TEMPLATE/bug_report.yml:81]: As discussed we are keeping this in for now, will be added to debug info for next release

File: .github/ISSUE_TEMPLATE/bug_report.yml (hunk 3 above)
Comment (droidmonkey) [reply .github/ISSUE_TEMPLATE/bug_report.yml:14]: Good point, will do

File: .github/ISSUE_TEMPLATE/feature_request.yml (hunk 4 above)
Comment (droidmonkey) [reply .github/ISSUE_TEMPLATE/feature_request.yml:33]: It exists in the current template, this is the place for people to plead their case for why the feature should be worked on. Not all features are worth investing time in, but if the use case is compelling or can spur some other ideas I would like to hear them.

**Review by phoerious — APPROVED** [review]

**Pull request merged**